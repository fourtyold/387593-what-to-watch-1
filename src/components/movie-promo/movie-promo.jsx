import React, {Fragment} from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import MyListButton from "../my-list-button/my-list-button.jsx";
import PlayButton from "../play-button/play-button.jsx";
import {getPromoFilm} from "../../reducer/films/selectors.js";

const MoviePromo = (props) => {
  const {film, width, height} = props;
  return <Fragment>
    <div className="movie-card__info">
      <div className="movie-card__poster">
        <img src={film.posterImage} alt={`${film.name} poster`} width={width} height={height} />
      </div>

      <div className="movie-card__desc">
        <h2 className="movie-card__title">{film.name}</h2>
        <p className="movie-card__meta">
          <span className="movie-card__genre">{film.genre}</span>
          <span className="movie-card__year">{film.released}</span>
        </p>

        <div className="movie-card__buttons">
          <PlayButton id={film.id}/>
          <MyListButton
            id={film.id}
            isFavorite={film.isFavorite}
          />
        </div>
      </div>
    </div>
  </Fragment>;
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  film: getPromoFilm(state)
});

MoviePromo.propTypes = {
  film: PropTypes.shape({
    posterImage: PropTypes.string,
    name: PropTypes.string,
    genre: PropTypes.string,
    released: PropTypes.number,
    id: PropTypes.number
  }).isRequired,
  width: PropTypes.number,
  height: PropTypes.number
};

MoviePromo.defaultProps = {
  width: 218,
  height: 327
};

export {MoviePromo};

export default connect(mapStateToProps)(MoviePromo);
