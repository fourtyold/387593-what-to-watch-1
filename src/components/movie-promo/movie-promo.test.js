import React from "react";
import renderer from "react-test-renderer";
import {HashRouter} from "react-router-dom";
import {Provider} from "react-redux";
import {createStore} from "redux";

import {MoviePromo} from "./movie-promo.jsx";
import combineReducers from "../../reducer/index.js";

const options = {
  film: {
    posterImage: `poster`,
    name: `name`,
    genre: `genre`,
    released: 2000,
    id: 1
  },
  width: 100,
  height: 300
};

it(`Movie promo correctly renders`, () => {
  const tree = renderer
    .create(
        <Provider store={createStore(combineReducers)}>
          <HashRouter>
            <MoviePromo
              film={options.film}
              width={options.width}
              height={options.height}
            />
          </HashRouter>
        </Provider>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
