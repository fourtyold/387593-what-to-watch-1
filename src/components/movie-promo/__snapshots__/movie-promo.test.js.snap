// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie promo correctly renders 1`] = `
<div
  className="movie-card__info"
>
  <div
    className="movie-card__poster"
  >
    <img
      alt="name poster"
      height={300}
      src="poster"
      width={100}
    />
  </div>
  <div
    className="movie-card__desc"
  >
    <h2
      className="movie-card__title"
    >
      name
    </h2>
    <p
      className="movie-card__meta"
    >
      <span
        className="movie-card__genre"
      >
        genre
      </span>
      <span
        className="movie-card__year"
      >
        2000
      </span>
    </p>
    <div
      className="movie-card__buttons"
    >
      <a
        className="btn btn--play movie-card__button"
        href="#/film/1/player"
        onClick={[Function]}
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </a>
      <button
        className="btn btn--list movie-card__button"
        type="button"
      >
        <svg
          height="20"
          viewBox="0 0 19 20"
          width="19"
        >
          <use
            xlinkHref="#add"
          />
        </svg>
        <span>
          My list
        </span>
      </button>
    </div>
  </div>
</div>
`;
